File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1277.cpp
#include <iostream>
#include <vector>
#include <utility>

using namespace std;

int main(){
    int n;
    cin >> n;
    
    while(n--){
        int v;
        cin >> v;
        
        string nome[110], freq[110];
        
        for(int i = 0; i < v; i++)
            cin >> nome[i];
        
        for(int i = 0; i < v; i++)
            cin >> freq[i];
        
        
        vector<string> falt;
        for(int i = 0; i < v; i++){
            
            double f = 0, p = 0;
            for(int j = 0; j < freq[i].size(); j++){
                if(freq[i][j] == 'P') p++;
                else if(freq[i][j] == 'A') f++;
            }
            
            if((p * 100) / (p + f) < 75) falt.push_back(nome[i]);
            
            
        }
        
        for(int j = 0; j < falt.size(); j++){
            cout << falt[j];
            
            if(j != falt.size()-1) cout << " ";
        }
        cout << endl;
        
    }
    
    return 0;
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1277.py
n = int(input())

for i in range(n):
    k = int(input())
    
    nomes = input().split()
    freq = input().split()
    
    falt = []
    for j in range(k):
        p = 0
        f = 0
        
        for let in freq[j]:
            if(let == 'P'): p += 1
            elif(let == 'A'): f += 1
        
        if((p*100) / (p+f) < 75): falt.append(nomes[j])
    
    for i in range(len(falt)):
        print(falt[i], end = '')
        if(i != len(falt)-1): print(' ', end = '')
    
    print()